src/main/c/powerwaf_jni.c

```diff
@@ -595,6 +595,9 @@ static jobject _run_rule_common(bool is_byte_buffer, JNIEnv *env, jclass clazz,
     if (ctx) {
         ddwaf_context_destroy(ctx);
     }
+    if (!is_byte_buffer) {
+        ddwaf_object_free(&input);
+    }
```

Comment:
I think this is correct, but only because you pass the parameters as ephemeral. In general, ddwaf will take care of destroying pameters passed to ddwaf_run after `ddwaf_context_destroy` (if [a free function is given](https://github.com/DataDog/libddwaf-java/blob/e4fc3dbad380797bdcc1b290daa6a3c55b85b5a5/src/main/c/powerwaf_jni.c#L2226)), so this would effectively be a double free in the normal path. But confirm with @Anilm3.

Reply:
yeah, so you need to call `ddwaf_object_free` on `input` only if `!is_byte_buffer` and one of a) we bailed out before calling `ddwaf_run` or b) `ddwaf_run` returns DDWAF_ERR_INVALID_ARGUMENT

Reply:
Ok, i fixed it in [aa0072a](https://github.com/DataDog/libddwaf-java/pull/95/commits/aa0072a82f7701d6f06acd836525625b696e9265)

Reply:
Thanks